samples/corda/corda-simple-application/clients/src/main/kotlin/com/cordaSimpleApplication/client/AssetTransferManager.kt

```diff
@@ -0,0 +1,705 @@
+                val assetPledgeStatusBytes = proxy.startFlow(::GetAssetPledgeStatusByPledgeId, pledgeId!!, recipientNetworkId!!)
+                    .returnValue.get()
+                val charset = Charsets.UTF_8
+                println("assetPledgeStatus: ${assetPledgeStatusBytes.toString(charset)}")
+                println("assetPledgeStatus: ${assetPledgeStatusBytes.contentToString()}")
+            } catch (e: Exception) {
+                println("Error: ${e.toString()}")
+            } finally {
+                rpc.close()
+            }
+        }
+    }
+}
+
+private fun getReclaimViewAddress(
+    assetCategory: String,
+    assetType: String,
+    assetIdOrQuantity: String,
+    pledgeId: String,
+    pledgerCert: String,
+    exportNetworkId: String,
+    recipientCert: String,
+    importNetworkId: String,
+    pledgeExpiryTimeSecs: String) : String
+{
+    var viewAddress: String
+    var funcName: String? = null
+    var funcArgs: List<String>? = null
+
+    if (assetCategory.equals("token")) {
+        if (importNetworkId.equals("Corda_Network") || importNetworkId.equals("Corda_Network2")) {
+            funcName = "GetAssetClaimStatusByPledgeId"
+            funcArgs = listOf(pledgeId, pledgeExpiryTimeSecs)
+        } else if (importNetworkId.equals("network1") || importNetworkId.equals("network2")) {
+            funcName = "GetTokenAssetClaimStatus"
+            funcArgs = listOf(pledgeId, assetType, assetIdOrQuantity, recipientCert, pledgerCert,
+                                exportNetworkId, pledgeExpiryTimeSecs)
+        }
+    }
+    println("funcName: $funcName and funcArgs: $funcArgs")
+
+    viewAddress = generateViewAddressFromRemoteConfig(importNetworkId, funcName!!, funcArgs!!)
+
+    return viewAddress
+}
+
+private fun getClaimViewAddress(
+    assetCategory: String,
+    pledgeId: String,
+    pledgerCert: String,
+    exportNetworkId: String,
+    recipientCert: String,
+    importNetworkId: String) : String
+{
+    var viewAddress: String
+    var funcName: String? = null
+    var funcArgs: List<String>? = null
+
+    if (assetCategory.equals("token")) {
+        if (exportNetworkId.equals("Corda_Network") || exportNetworkId.equals("Corda_Network2")) {
+            funcName = "GetAssetPledgeStatusByPledgeId"
+            funcArgs = listOf(pledgeId, importNetworkId)
+        } else if (exportNetworkId.equals("network1") || exportNetworkId.equals("network2")) {
+            funcName = "GetTokenAssetPledgeStatus"
+            funcArgs = listOf(pledgeId, pledgerCert, importNetworkId, recipientCert)
+        }
+    }
+    println("funcName: $funcName and funcArgs: $funcArgs")
+
+    viewAddress = generateViewAddressFromRemoteConfig(exportNetworkId, funcName!!, funcArgs!!)
+
+    return viewAddress
+}
+
+/*
+ * This function parses the file remote-network-config.json and fetches the values for keys of an input network.
+ * If either the network is not found, or the key is not found, it throws an exception.
+ */
```

Comment:
Is this the Corda equivalent of the Fabric CLIs (Node or Go)?

Reply:
Yes, Rama.